PasswordVerifier/src/utils.java

```diff
@@ -0,0 +1,104 @@
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutput;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+
+public class utils {
+	public static int MIN_LENGTH = 8;
+	
+	/*
+	 * contains logic for which characters are allowed
+	 * @param c character to be checked
+	 * @return true if c is allowed, false otherwise
+	 */
+	private static boolean isValidChar(char c) {
+		if (c < '!' || c > '~')
+			return false;
+		return true;
+	}
+	
+	/*
+	 * checks if pwd is a valid password according to the following rules:
+	 * at least one lowercase letter, uppercase letter, number, and special
+	 * character, and all characters must be valid as defined in isValidChar
+	 * @param pwd password to be verified
+	 * @return true if pwd is valid, false otherwise
+	 */
+	public static boolean isValid(String pwd) {
+		if (pwd.length() < MIN_LENGTH)
+			return false;
+		boolean hasLower = false;
+		boolean hasUpper = false;
+		boolean hasNumber = false;
+		boolean hasSpecial = false;
+		char current;
+		for (int i = 0; i < pwd.length(); ++i) {
+			current = pwd.charAt(i);
+			if(!utils.isValidChar(current))
```

Comment:
Nit: I've always made a practice of having spaces around each keyword, so if (...), this may be something an auto linter could do